src/c++/library/http_client.cc

```diff
@@ -561,9 +561,9 @@ HttpInferRequest::GetNextInput(uint8_t* buf, size_t size, size_t* input_bytes)
 
       data_buffers_.front().first += csz;
       data_buffers_.front().second -= csz;
-      if (data_buffers_.front().second == 0) {
-        data_buffers_.pop_front();
-      }
+    }
```

Comment:
Old code would only pop the empty data buffer if it started not-empty. This lead to an infinite loop if an empty data buffer was passed in. Now it will pop the empty data buffer once it is empty, regardless of how it started.

Reply:
Great find!